src/components/_helpers/transferProps.js

```diff
@@ -1,7 +1,46 @@
-export const transferProps = ({
-  children,
-  className,
-  ref,
-  staticContext,
-  ...restProps
-}) => restProps;
+/**
+ * Controls passing of props from the React component to the HTML element
+ *
+ * Sometimes it is useful to have a mechanism to pass props from the React component to a rendered HTML element.
+ * It enables making the component interactive and helps improve its accessibility. However some props should
+ * never be passed to the HTML element as it would break things. This function is used to filter out such props.
+ *
+ * When run in development mode, the function will log the error to the console if any invalid props are passed.
+ *
+ * @param props The props that were passed to the React component
+ * @returns The props to be passed to the HTML element
+ */
+export const transferProps = (props) => {
+  const {
+    children,
+    className,
+    contentEditable,
+    dangerouslySetInnerHTML,
+    ref,
+    staticContext,
+    style,
+    suppressContentEditableWarning,
+    ...restProps
+  } = props;
+
+  if (process.env.NODE_ENV !== 'production') {
+    const invalidProps = Object.keys({
```

Comment:
I would rather keep this as array of strings due to performance. And this array should be as constant before the function to prevent inicialization on every function call.

Reply:
I changed the implementation.

Since the code only runs in dev mode I don't think it makes much of a difference, but it for sure does not hurt.